gdnative-derive/src/lib.rs

```diff
@@ -16,6 +16,43 @@ mod profiled;
 mod varargs;
 mod variant;
 
+/// Collects method signatures of all functions in a `NativeClass` that have the `#[export]` attribute and registers them with Godot.
+///
+/// For example, in the following class
+/// ```ignore
+/// #[derive(NativeClass)]
+/// #[inherit(Reference)]
+/// #[no_constructor]
+/// struct Foo {}
+///
+/// #[methods]
+/// impl Foo {
+///     #[export]
+///     fn foo(&self, _owner: &Reference, bar: i64) -> i64 {
+///         bar
+///     }
+/// }
+///
+/// ```
+/// Will expand to 
+/// ```ignore
+/// struct Foo{}
+/// impl gdnative::nativescript::NativeClassMethods for Foo {
+///     fn register(builder: &ClassBuilder<Self>) {
+///         use gdnative::nativescript::init::*;
+///         builder.build_method("foo", gdnative::godot_wrap_method!(Foo::foo))
+///             .with_rpc_mode(Rpc::Disabled)
+///             .done_stateless();
+///     }
+///
+///     fn foo(&self, _: &Reference, bar: i64) -> i64 {
+///         bar
+///     }
+/// }
+/// ```
+/// **Important**: Only one `impl` block per struct may be attributed with `#[methods]`.
+///
+/// For more context, please refer to [gdnative::NativeClass](NativeClass)
```

Comment:
Generally, I would add full-stops to all sentences when the documentation is a bigger block.